### ISSUE TITLE
does not seem to support the sequence * operator

### ISSUE THREAD

**Issue by zzzeek** [microsoft/pyright#3037]
I searched through all known docs and github issues for both pylance and pyright to see if this is a known limitation but couldn't find anything.   pyright does not recognize the "*" operator as the [sequence multiplication operator](https://docs.python.org/3/library/stdtypes.html#common-sequence-operations).

pyright 1.1.221

```py
x = 8 * "x"

y = "y" * 8

z = [1, 2, 3] * 4

print(x)
print(y)
print(z)

```

```
/home/classic/dev/sqlalchemy/test3.py
  /home/classic/dev/sqlalchemy/test3.py:1:5 - error: Operator "*" not supported for types "Literal[8]" and "Literal['x']"
    Operator "*" not supported for types "Literal[8]" and "Literal['x']" (reportGeneralTypeIssues)
  /home/classic/dev/sqlalchemy/test3.py:3:5 - error: Operator "*" not supported for types "Literal['y']" and "Literal[8]"
    Operator "*" not supported for types "Literal['y']" and "Literal[8]" (reportGeneralTypeIssues)
  /home/classic/dev/sqlalchemy/test3.py:5:5 - error: Operator "*" not supported for types "list[int]" and "Literal[4]"
    Operator "*" not supported for types "list[int]" and "Literal[4]" (reportGeneralTypeIssues)
3 errors, 0 warnings, 0 informations 
```

the program's output is:

```
$ python test3.py 
xxxxxxxx
yyyyyyyy
[1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3]
```

**Comment by erictraut** [comment]
Hmm, I'm not able to repro this problem. Are you using a version of the typeshed stubs other than the ones bundled with pyright?

Are you using pyright (or pylance) as a language server in VS Code or some other editor? If so, could you try the following experiment?

Create a new python source file (e.g. test.py) and paste the following:
```python
a = "x"
b = a.__rmul__(8)
```

Then right click on `__rmul__` and choose "Go To Declaration". This should take you to the typeshed class definition for `str`, specifically to a method with a signature like this:
`def __rmul__(self, n: SupportsIndex) -> str: ...`

**Comment by zzzeek** [comment]
OK it shows the file as:

/home/classic/.vscode/extensions/ms-python.vscode-pylance-2022.2.1-pre.1/dist/typeshed-fallback/stdlib/builtins.pyi

looks like this:

```py
class str(Sequence[str]):

     # ... other methods omitted

    def __rmul__(self, n: SupportsIndex) -> str: ...

```

SupportsIndex seems to be coming from some other place totally, from my Python venv:

/home/classic/.venv3100/lib/python3.10/site-packages/typing_extensions-stubs/__init__.pyi

```py
@runtime_checkable
class SupportsIndex(Protocol, metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def __index__(self) -> int: ...

```

**Comment by erictraut** [comment]
Ah, that's the problem. You have a really old `typing_extensions` stub package installed, and it's overriding the newer version. You should `pip uninstall typing_extensions-stubs`. That should clear up your problem.

**Comment by zzzeek** [comment]
it seems to be this:

    pip uninstall types_typing_extensions

but yes that was it thanks

**Comment by erictraut** [comment]
@JelleZijlstra, this is the second or third person who has run into this issue recently. Do you have any suggestions for how we can help people avoid this? The stdlib stubs (like `builtins.pyi` import symbols from `typing_extensions`, so this incompatibility is likely to bite a lot of users.

**Comment by JelleZijlstra** [comment]
Oh, that's unfortunate. I think when we did the typeshed redesign we briefly treated typing-extensions as a third-party package before we changed our mind and decided to pretend it's part of the stdlib.

Here's a couple of (non-exclusive) ideas:
- In type checkers, add a special case that ignores stubs installed as `types-typing-extensions`, and/or warns people if the type checker detects that it is installed. 
- yank all releases of `types-typing-extensions` from PyPI. But that won't help people who already have it installed.
- put up a new release of `types-typing-extensions` that tells you not to use it. Again, that doesn't help people who install an older version.

**Comment by erictraut** [comment]
This wasn't an issue until recently because pyright previously always resolved its bundled stdlib typeshed stubs first, so no one could override them. But it was pointed out that pyright wasn't properly conforming with PEP 561, so we decided to change it. 

I think it's a good idea to yank releases of `types-typing-extensions` from pypi so the problem doesn't get worse over time.

**Comment by JelleZijlstra** [comment]
Unfortunately I can't do that from my PyPI account, since only the bot owns the types-* projects. Could you open an issue on typeshed about this so we can figure out how to do iit? Sebastian may have some ideas too.

**Comment by erictraut** [comment]
OK, I'll do that.
